app/models/duration_rate.rb

```diff
@@ -6,12 +6,13 @@ class DurationRate < ApplicationRecord
 
   belongs_to :price_policy, required: true
 
-  validate :rate_or_subsidy
-  validates :rate, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
+  validates :rate, presence: true
+  validates :rate, numericality: { greater_than_or_equal_to: 0 }
+  validates :subsidy, presence: true, if: -> { price_policy.price_group.is_internal? && !price_policy.price_group.master_internal? }
```

Comment:
I think you can use the `price_group` method (line 61):
```suggestion
  validates :subsidy, presence: true, if: -> { price_group.is_internal? && !price_group.master_internal? }
```
... I think this is the 3rd validation where we need to check for this so maybe it's time for a method like `requires_subsidy?`.